agent/task_context.go

```diff
@@ -30,7 +30,16 @@ func (tc *taskContext) getCurrentCommand() command.Command {
 	return tc.currentCommand
 }
 
-func (tc *taskContext) setCurrentIdleTimeout(cmd command.Command) {
+// setCurrentIdleTimeout sets the idle timeout for the current running command.
+// This timeout only applies to commands running in specific blocks where idle
+// timeout is allowed.
+func (tc *taskContext) setCurrentIdleTimeout(cmd command.Command, block command.BlockType) {
+	switch block {
+	case command.PreBlock, command.SetupGroupBlock, command.SetupTaskBlock, command.MainTaskBlock:
+	default:
+		return
```

Comment:
Should this return an error so that it doesn't result in progammer error down the road?

Reply:
`setCurrentIdleTimeout` is called even for commands blocks where idle timeout doesn't apply. I changed it to move this block check to the outer function.